docs/source/assets_writer.rst

```diff
@@ -0,0 +1,103 @@
+.. Copyright 2016 The Cartographer Authors
+
+.. Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+..      http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+.. cartographer SHA: 30f7de1a325d6604c780f2f74d9a345ec369d12d
+.. cartographer_ros SHA: 44459e18102305745c56f92549b87d8e91f434fe
+
+
+Assets writer
+=============
+
+The purpose of SLAM is to compute the trajectory of a single sensor through a metric space.
+On a higher level, the input of SLAM is sensor data, its output is the best estimate of the trajectory up to this point in time.
+To be real-time and efficient, Cartographer throws most of the sensor data away immediately.
+
+The trajectory alone is rarely of interest.
+But once the best trajectory is established, the full sensor data can be used to derive and visualize information about its surroundings.
+
+Cartographer provides the assets writer for this.
```

Comment:
@msy22 Thank you very much for the review. This was tremendously useful and I adopted most of your suggestions!

It is hard to write documentation for me since I am so well acquainted with code base, nomenclature and design, so adapting a beginners mind is nearly impossible. Having an interested outside party like yourself reading the documentation provides very useful context and is a tremendous value add.